core/arch/arm/arm.mk

```diff
@@ -65,8 +65,16 @@ CFG_CORE_RODATA_NOEXEC ?= n
 ifeq ($(CFG_CORE_RODATA_NOEXEC),y)
 $(call force,CFG_CORE_RWDATA_NOEXEC,y)
 endif
+
 # 'y' to set the Alignment Check Enable bit in SCTLR/SCTLR_EL1, 'n' to clear it
-CFG_SCTLR_ALIGNMENT_CHECK ?= y
+# Saying 'n' here allows to run EL0 code built without -mstrict-align (Aarch64)
+# or -mno-unaligned-access (Aarch32). One notable example is the GCC C++
+# runtime libraries in the standard Linux cross-compilers (aarch64-linux-gnu
+# and arm-linux-gnueabihf).
+# EL1 code is always built with strict alignment because some parts access
+# device memory or run when the MMU is disabled. In both cases unaligned
+# accesses are forbidden.
+CFG_SCTLR_ALIGNMENT_CHECK ?= n
```

Comment:
Comment for "Disable alignment check by default"

Generally I agree, but I think that our CI testing should still use this.

Reply:
You mean build-tested (Shippable) or runtime-tested (IBART)?

Reply:
Enabling it on IBART would disable C++ tests on IBART. I guess the benefits of `CFG_SCTLR_ALIGNMENT_CHECK=y` is just to smoke out eventual minor inefficiencies so compared to C++ it's rather low prio.

Let's just maintain two targets (one arm32 and one arm64) on shippable with this.

Reply:
OK, fixed